### ISSUE TITLE
Verify that pycoreir works with coreir branch "configs"

### ISSUE THREAD

**Issue by rdaly525** [leonardt/pycoreir#7]
This branch changed "Param" to be a real C++ object called ValueType. I suspect the API will need to be updated accordingly. I also am using Dillon's dynamic_bit_vector class in order to do sized ints. some part of that api may also need to be ported.

I also have a few naming changes. 
in my C++ code:
```
s/Const(/Const::make(c,/
s/Args /Values /
s/AINT/c->Int()/
s/ABOOL/c->Bool()/
s/ASTRING/c->String()/
s/Config/Mod/
```

**Comment by leonardt** [comment]
Branch https://github.com/leonardt/pycoreir/tree/configs works in my local environment after yesterday's changes.  Fails on travis due to the same error as the coreir configs branch, will merge once that's resolved and configs is merged into coreir dev.
